Round corners beneath menu bar on primary monitor

RdStudios9145 opened the pull request:
Rounds the corners beneath the menu bar on the primary monitor. Also adds rounds around the menu bar for opening the applications menu.

RdStudios9145 committed:
Round corners beneath menu bar on primary monitor

lennart-k commented:
Hey, thanks for your contribution!
However, at the moment there are still some problems:
- The top bar can have a different height than 32px (mine is smaller)
- If the top bar is transparent we must not show these corners
- The corners would appear over many fullscreen apps
- code duplication (double bottom corners, I could take care of that)
- For the above reasons we'd also need a setting to disable the extra corners

RdStudios9145 committed:
added settings

RdStudios9145 committed:
added settings

RdStudios9145 committed:
struggling with git

RdStudios9145 committed:
More git struggles

RdStudios9145 committed:
Schema contained spaces

RdStudios9145 commented:
Fixed the hard-coded width and added a setting for enabling the extra corners.

The bottom left and right are not duplicates as they are also placed around the menu bar, and only appears when opening the apps menu. 
![image](https://github.com/user-attachments/assets/a638a477-2015-4adb-912f-6aced7eac3ea)

I'm not worried about them appearing over apps as the main ones already do, this will only make it more seamless for the primary monitor. I don't know how to get the transparency or color of the main menu bar, so it could be disabled through settings (also why I added the extra bottom-style corners in the main menu bar)

lennart-k commented:
Thanks for the follow-up and your effort (and the screenshot).
- The problem with fullscreen apps is that they take the full screen so also the top bar space
- Also I'm not too happy about how they look like when the top bar is not black (e.g. in the window overview)
  - that will also look weird when the corner radius is larger than the top bar height
- One needs to hide them dynamically which I feel is quite a difficult task with all edge cases. (and people will be complaining about the feature not working :D)

For these reasons (especially the latter) and given that I don't really have the time to fix these problems I don't feel comfortable with merging this feature, sorry :/

I still wish you lots of fun on your desktop customisation journey :)
